scripts/modtools/extra-gamelog.lua

```diff
@@ -0,0 +1,192 @@
+    if (not snowing and not raining) then msg("The weather has cleared.")
+    elseif raining then msg("It has started raining.")
+    elseif snowing then msg("A snow storm has come.")
+    end
+
+    -- Log site information for forts
+    if df.world_site.find(df.global.ui.site_id) == nil then return end
+    local site = df.world_site.find(df.global.ui.site_id)
+    local fort_ent = df.global.ui.main.fortress_entity
+    local civ_ent = df.historical_entity.find(df.global.ui.civ_id)
+    local function fullname(item)
+        return dfhack.TranslateName(item.name)..' ('..dfhack.TranslateName(item.name ,true)..')'
+    end
+    msg('Loaded '..df.global.world.cur_savegame.save_dir..', '..fullname(df.global.world.world_data)..
+        ' at coordinates ('..site.pos.x..','..site.pos.y..')')
+    msg('Loaded the fortress '..fullname(site)..
+        (fort_ent and ', colonized by the group '..fullname(fort_ent) or '')..
+        (civ_ent and ' of the civilization '..fullname(civ_ent)..'.' or '.'))
+end
+
+
+old_expedition_leader = nil
+old_mayor = nil
+function log_nobles()
+    local expedition_leader = nil
+    local mayor = nil
+    local function check(unit)
+        if not dfhack.units.isCitizen(unit) then return end
+        for _, pos in ipairs(dfhack.units.getNoblePositions(unit) or {}) do
+            if pos.position.name[0] == "expedition leader" then
+                expedition_leader = unit
+            elseif pos.position.name[0] == "mayor" then
+                mayor = unit
+            end
+        end
+    end
+    for _, unit in ipairs(df.global.world.units.active) do
+        check(unit)
+    end
+
+    if old_mayor == nil and expedition_leader == nil and mayor ~= nil and old_expedition_leader ~= nil then
+        msg("Expedition leader was replaced by mayor.")
+    end
+
+    if expedition_leader ~= old_expedition_leader then
+        if expedition_leader == nil then
+            msg("Expedition leader position is now vacant.")
+        else
+            msg(dfhack.TranslateName(dfhack.units.getVisibleName(expedition_leader)).." became expedition leader.")
+        end
+    end
+
+    if mayor ~= old_mayor then
+        if mayor == nil then
+            msg("Mayor position is now vacant.")
+        else
+            msg(dfhack.TranslateName(dfhack.units.getVisibleName(mayor)).." became mayor.")
+        end
+    end
+    old_mayor = mayor
+    old_expedition_leader = expedition_leader
+end
+
+siege = false
+function log_siege()
+    local function cur_siege()
+        for _, unit in ipairs(df.global.world.units.active) do
+            if unit.flags1.active_invader then return true end
+        end
+        return false
+    end
+    local old_siege = siege
+    siege = cur_siege()
+    if siege ~= old_siege and siege then
+        msg("A vile force of darkness has arrived!")
+    elseif siege ~= old_siege and not siege then
+        msg("Siege was broken.")
+    end
+end
+
+
+local workshopTypes = {
+    [0]="Carpenters Workshop",
```

Comment:
This (and the leadership logging above) is taken from the `soundsense` script Zwei distriubutes - I added most of the logging it does, somewhat more structured and without the spammy bits.